src/DTO/UserDTO.php

```diff
@@ -16,86 +20,150 @@ class UserDTO
     /**
      * @var integer
      * @Serializer\Type("integer")
+     * @Serializer\Expose()
+     * @Groups({"UserCreate", "UserEdit", "PasswordEdit"})
      */
     public $id;
 
     /**
      * @var string
      * @Serializer\Type("string")
-     * @Assert\NotNull
+     * @Assert\NotNull(groups={"UserCreate", "UserEdit"})
      * @Assert\Length(
      *      min = 4,
      *      max = 50,
      *      minMessage = "Your first name must be at least 4 characters long",
-     *      maxMessage = "Your first name cannot be longer than 50 characters"
+     *      maxMessage = "Your first name cannot be longer than 50 characters",
+     *     groups={"UserCreate", "UserEdit"}
      * )
+     * @Serializer\Expose()
+     * @Groups({"UserCreate", "UserEdit"})
      */
     public $username;
 
     /**
      * @var string
      * @Serializer\Type("string")
-     * @Assert\NotNull
+     * @Serializer\Expose()
+     * @Groups({"UserCreate", "PasswordEdit"})
+     * @Assert\NotNull(groups={"UserCreate", "PasswordEdit"})
      * @Assert\Regex("/^\S*(?=\S{8,})(?=\S*[a-z])(?=\S*[A-Z])(?=\S*[\d])(?=\S*[-_!@#$%^&*])\S*$/",
-     * message = "password requirements(at least):length >8, 1 uppercase, 1 lowercase, 1 digit, 1 special"
+     * message = "password requirements(at least):length >8, 1 uppercase, 1 lowercase, 1 digit, 1 special",
+     * groups={"UserCreate", "PasswordEdit"}
      * )
      */
     public $password;
 
     /**
      * @var string
      * @Serializer\Type("string")
-     * @Assert\NotNull
+     * @Assert\NotNull(groups={"PasswordEdit"})
+     * @Assert\Regex("/^\S*(?=\S{8,})(?=\S*[a-z])(?=\S*[A-Z])(?=\S*[\d])(?=\S*[-_!@#$%^&*])\S*$/",
+     * message = "password requirements(at least):length >8, 1 uppercase, 1 lowercase, 1 digit, 1 special",
+     * groups={"PasswordEdit"}
+     * )
+     * @Serializer\Expose()
+     * @Groups({"PasswordEdit"})
+     */
+    public $newPassword;
+
+    /**
+     * @var string
+     * @Serializer\Type("string")
+     * @Assert\NotNull(groups={"UserCreate"})
+     * @Assert\EqualTo(propertyPath="password",
+     *     message="Passwords do not match.",
+     *     groups={"UserCreate"}
+     *     )
+     * @Serializer\Expose()
+     * @Groups({"UserCreate"})
+     */
+    public $confirmPassword;
+
+    /**
+     * @var string
+     * @Serializer\Type("string")
+     * @Assert\NotNull(groups={"PasswordEdit"})
+     * @Assert\EqualTo(propertyPath="newPassword",
+     *     message="Passwords do not match.",
+     *     groups={"PasswordEdit"}
+     *     )
+     * @Serializer\Expose()
+     * @Groups({"PasswordEdit"})
+     */
+    public $confirmNewPassword;
```

Comment:
You can rename the fields so you don't have 2 confirm fields. On register you would have password, confirmPassword and on password change you would have oldPassword, password, confirmPassword. The validation rules will also be simpler in this case.